File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/StatusApiException.java
package com.github.seratch.jslack.api.status.v2;

import net.socialhub.http.HttpResponse;

public class StatusApiException extends Exception {

    private final HttpResponse response;
    private final String responseBody;

    public StatusApiException(HttpResponse response, String responseBody) {
        this.response = response;
        this.responseBody = responseBody;
    }

    public HttpResponse getResponse() {
        return this.response;
    }

    public String getResponseBody() {
        return this.responseBody;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/status/StatusApiException.kt
package work.socialhub.kslack.api.status

import work.socialhub.khttpclient.HttpResponse

/**
 * Exception thrown when the Slack Status API returns an error response.
 *
 * Wraps the HTTP response and raw response body for error diagnosis
 * when querying the public Slack Status page.
 *
 * @param response The HTTP response that triggered the exception
 * @param responseBody The raw response body text
 */
class StatusApiException(
    val response: HttpResponse,
    val responseBody: String
) : Exception() {

}
